include/oneapi/dpl/pstl/execution_impl.h

```diff
@@ -29,100 +30,131 @@ namespace dpl
 namespace __internal
 {
 
-/* predicate */
+//------------------------------------------------------------------------
+// backend selector with tags
+//------------------------------------------------------------------------
+
+#if _ONEDPL_PAR_BACKEND_TBB
+using __par_backend_tag = __tbb_backend_tag;
+#elif _ONEDPL_PAR_BACKEND_OPENMP
+using __par_backend_tag = __omp_backend_tag;
+#elif _ONEDPL_PAR_BACKEND_SERIAL
+using __par_backend_tag = __serial_backend_tag;
+#else
+#    error "Parallel backend was not specified"
+#endif
+
+template <class _IsVector>
+struct __serial_tag
+{
+    using __is_vector = _IsVector;
+    using __backend_tag = __par_backend_tag;
```

Comment:
Just one naive question. Why de we need `__backend_tag ` type for the `__serial_tag` tag?

Reply:
We need it to specialize __buffer class with proper backend tag in the places where we have then one type oh host dispatch tags and `__serial_tag` too.
For example, it's in the function:
```C++
template <class _Tag, typename Policy, typename InputIterator1, typename InputIterator2, typename OutputIterator,
          typename T, typename BinaryPredicate, typename Operator>
OutputIterator
pattern_exclusive_scan_by_segment(_Tag, Policy&& policy, InputIterator1 first1, InputIterator1 last1,
                                  InputIterator2 first2, OutputIterator result, T init, BinaryPredicate binary_pred,
                                  Operator binary_op)
{
    static_assert(__internal::__is_host_dispatch_tag_v<_Tag>);

    using __backend_tag = typename _Tag::__backend_tag;

    // ...
    oneapi::dpl::__par_backend::__buffer<__backend_tag, Policy, FlagType> _flags(policy, n);
}
```

Reply:
Fixed: added new trait
```C++
template <class _Tag, typename = void>
struct __buffer_backend_tag_type
{
    using type = typename _Tag::__backend_tag;
};

template <class _Tag>
struct __buffer_backend_tag_type<_Tag, ::std::enable_if_t<__is_serial_tag_v<_Tag>>>
{
    using type = __par_backend_tag;
};
```